crates/bevy_asset/src/assets.rs

```diff
@@ -509,13 +543,13 @@ impl<A: Asset> Assets<A> {
                         },
                         marker: PhantomData,
                     };
-                    (id, v)
+                    (id, v.clone())
```

Comment:
Technically, there's some possible contention on the refcount when cloning, but by not cloning, we incur a double dereference whenever accessing the Arc. Generally I prefer to just clone an `Arc` since what you really care about is accessing the data, not the container. A borrow to an `Arc` is just adding another layer of indirection. Having no lifetime on the iterator item is nice!

Reply:
I didn't consider the double deref, although I think you'd still be going through the first deref anyway if you always cloned the Arc. I agree that having ownership of the Arc is nicer for the user though, so I'm happy with `Arc<A>`.

Doesn't this PR *add* a lifetime to the iterator item, with the `+ '_`?
```diff
-    pub fn iter(&self) -> impl Iterator<Item = (AssetId<A>, &A)> {
+    pub fn iter(&self) -> impl Iterator<Item = (AssetId<A>, Arc<A>)> + '_ {
```
Unless I'm misunderstanding something :sweat_smile:

Reply:
This PR "adds" a lifetime to the *iterator* not the iterator item. Note the previous implementation also had this same lifetime, it was just elided since the return type includes a borrow. So we're just making the lifetime explicit. The difference for the new implementation is that we get ownership of the Arc like you mentioned.

Reply:
Ah, the lifetime was elided because of the `&A`, makes sense. I get it now.